### ISSUE TITLE
Allow set startTime and endTime in CloudWatch(streaming mode) datasource

### ISSUE THREAD

**Issue by lucgiang-novobi** [crowdsecurity/crowdsec#1953]
### What would you like to be added?

For feature request please pick a kind label by removing `<!-- -->` that wrap the example lines below

/kind enhancement
When CrowdSec start reading from CloudWatch logs stream, it read entire log events(from earliest events). It can make duplicated issues when we restart a CrowdSec container with same configuration. Should we have limit [startTime](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_GetLogEvents.html#API_GetLogEvents_RequestSyntax) and [endTime](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_GetLogEvents.html#API_GetLogEvents_RequestSyntax) parameters in [GetLogEventsPagesWithContext](https://github.com/crowdsecurity/crowdsec/blob/4a6a9c435583fc55a5177d7f098222e2903a27c5/pkg/acquisition/modules/cloudwatch/cloudwatch.go#L458)
```go
  err := cw.cwClient.GetLogEventsPagesWithContext(ctx,
	  &cloudwatchlogs.GetLogEventsInput{
		  Limit:         aws.Int64(cfg.GetLogEventsPagesLimit),
		  LogGroupName:  aws.String(cfg.GroupName),
		  LogStreamName: aws.String(cfg.StreamName),
		  NextToken:     startFrom,
		  StartFromHead: aws.Bool(true),
	  },
```

### Why is this needed?

To avoid reading entirely large CloudWatch log stream when use CloudWatch datasource

**Comment by lucgiang-novobi** [comment]
Other solution, could we store CloudWatch Stream token into file, and load this file before starting acquisition modules.
[CW stream token varible](https://github.com/crowdsecurity/crowdsec/blob/4a6a9c435583fc55a5177d7f098222e2903a27c5/pkg/acquisition/modules/cloudwatch/cloudwatch.go#L441)
```go
	streamIndexMutex.Lock()
	v := cw.streamIndexes[cfg.GroupName+"+"+cfg.StreamName] # store this variable into file
	streamIndexMutex.Unlock()
```

**Comment by buixor** [comment]
Hello @lucgiang-novobi !

Thanks for the report. Storing a token right now wouldn't be easy on the agent (no other datasource does it).
Would "simply" not starting from head every time do the trick? (ie. setting `StartFromHead` to false ?)

**Comment by lucgiang-novobi** [comment]
> Hello @lucgiang-novobi !
> 
> Thanks for the report. Storing a token right now wouldn't be easy on the agent (no other datasource does it). Would "simply" not starting from head every time do the trick? (ie. setting `StartFromHead` to false ?)

Yes, we need a configuration parameter to decide whether CrowdSec reads logs from the first event or last specific duration (i.e 30 minutes before). Reading the entirety of the log events at the beginning is not effective because Crowdsec should work with log events in real-time (or near real-time).

**Comment by lucgiang-novobi** [comment]
Hi @buixor ,
Do you have any solution for this issue? Please let me know.
Thank you!
